src/controllers/reportingController.ts

```diff
@@ -124,4 +201,28 @@ export class ReportingController {
 			}
 		}
 	}
+
+	async getOrganisationChoicesForUser(user: any) {
+		return {
+			directOrganisation: user.organisationalUnit,
+			typeaheadOrganisations: await this.csrsService.getOrganisationalUnitsForUser(user)
+		}
+	}
+
+	async userCanAccessReportingForOrganisation(user: any, organisationId: any) {
+		let organisationalUnitsForUser = await this.csrsService.getOrganisationalUnitsForUser(user)
+		let organisationIdsForUser = organisationalUnitsForUser.map(org => org.id)
+		let userCanAccessOrganisation = organisationIdsForUser.includes(parseInt(organisationId))
+		return userCanAccessOrganisation
+	}
+
+	async getChildOrganisationIdsForOrganisationId(organisationId: number): Promise<number[]> {
```

Comment:
This won't cascade down to all of the children levels, only the first layer. You should be able to use flatmap for this